### PR TITLE
[Bots] Add IsBot() to methods in attack.cpp where applicable.

### PR DESCRIPTION
**Pull request by Aeadoin**
- Warrior Bots weren't receiving innate 5% Mitigation bonus
- Bots were ignoring Rule Combat:LevelToStopACTwinkControl
- Bots were not receiving stunbash_chance bonuses.
- Bots could not trigger Skill Procs (SPA 427, 429) from AA bonuses.

### CODE REVIEW

File: zone/attack.cpp [hunk 1]
```diff
@@ -3957,10 +3958,11 @@ void Mob::CommonDamage(Mob* attacker, int64 &damage, const uint16 spell_id, cons
 		if (attacker) {
 			if (skill_used == EQ::skills::SkillBash) {
 				can_stun = true;
-				if (attacker->IsClient())
+				if (attacker->IsClient() || attacker->IsBot() || attacker->IsMerc()) {
```
Comment (Akkadius) [inline zone/attack.cpp:3961]: This is starting to be verbose and common usage pattern

Let's create a shorthand to consolidate

```
attacker->IsOfClientBotMerc()
```

Comment (Aeadoin) [reply zone/attack.cpp:3961]: excellent, I was going to ask you or KK about this. I'll add it.

**Review by Kinglykrab — COMMENTED** [review]
If you pull master in to this, you can change all the conditions to use the new method you added.

File: zone/attack.cpp (hunk 1 above)
Comment (Kinglykrab) [reply zone/attack.cpp:3964]: Can fix this, too.

File: zone/attack.cpp [hunk 2]
```diff
@@ -5161,8 +5163,9 @@ void Mob::ApplyMeleeDamageMods(uint16 skill, int64 &damage, Mob *defender, Extra
 		dmgbonusmod += opts->melee_damage_bonus_flat;
 
 	if (defender) {
-		if (defender->IsClient() && defender->GetClass() == WARRIOR)
+		if ((defender->IsClient() || defender->IsBot() || defender->IsMerc()) && defender->GetClass() == WARRIOR) {
```
Comment (Kinglykrab) [inline zone/attack.cpp:5166]: Can fix this, too.

File: zone/effects.cpp [hunk 3]
```diff
@@ -57,20 +57,20 @@ int64 Mob::GetActSpellDamage(uint16 spell_id, int64 value, Mob* target) {
...
+	if (spell_id == SPELL_IMP_HARM_TOUCH && IsOfClientBot()) //Improved Harm Touch
 		value -= GetAA(aaUnholyTouch) * 450; //Unholy Touch
 
 		chance = RuleI(Spells, BaseCritChance); //Wizard base critical chance is 2% (Does not scale with level)
 		chance += itembonuses.CriticalSpellChance + spellbonuses.CriticalSpellChance + aabonuses.CriticalSpellChance;
 		chance += itembonuses.FrenziedDevastation + spellbonuses.FrenziedDevastation + aabonuses.FrenziedDevastation;
 
 	//Crtical Hit Calculation pathway
-	if (chance > 0 || ((IsClient() || IsBot()) && GetClass() == WIZARD && GetLevel() >= RuleI(Spells, WizCritLevel))) {
+	if (chance > 0 || IsOfClientBot() && GetClass() == WIZARD && GetLevel() >= RuleI(Spells, WizCritLevel)) {
```
Comment (Kinglykrab) [inline zone/effects.cpp:68]: Removed too many parentheses here.

File: zone/attack.cpp (hunk 2 above)
Comment (Aeadoin) [reply zone/attack.cpp]: Thanks!

File: zone/effects.cpp (hunk 3 above)
Comment (Aeadoin) [reply zone/effects.cpp]: Thanks!

**Review by Kinglykrab — APPROVED** [review]
:+1:

**Pull request merged**